File: pset1/credit.c
#include <cs50.h>
#include <stdio.h>

int main(void)
{
    long value = get_long("Number: ");
    int sum = 0, sum2 = 0, digit, count = 0, check;
    bool ok = 0;

    while (value)
    {
        digit = value % 10;
        count ++;

        if (!ok)
        sum += digit;

        else
        {
            digit *= 2;
            sum2 += digit % 10;
            digit /= 10;
            if (digit)
            sum2 += digit;
        }

        ok = 1 - ok;
        value /= 10;

        if (value >= 10)
        check = value;
    }
    sum += sum2;

    if (sum % 10 == 0)
    {
        if (count == 15 && (check == 34 || check == 37))
        printf("AMEX\n");

        else if (count == 16 && (check >= 51 && check <= 55))
        printf("MASTERCARD\n");

        else if ((count == 13 || count == 16) && (check / 10) % 10 == 4)
        printf("VISA\n");

        else
        {
         printf("INVALID\n");
        }
    }

    else printf("INVALID\n");
}

File: pset6/credit.py
from cs50 import get_string

number = get_string("Number: ")
length = len(number)
ok = 1
sum1 = 0
sum2 = 0
count = 0

for i in number[::-1]:
    c = int(i)
    if ok == 1:
        sum1 += c
    else:
        c *= 2
        sum2 += c % 10
        c = int(c / 10)
        if c > 0:
            sum2 += c
    ok = 1 - ok
sum1 += sum2

if sum1 % 10 == 0:
    if len(number) == 15 and (int(number[0:2]) == 34 or int(number[0:2]) == 37):
        print("AMEX")
    elif len(number) == 16 and (int(number[0:2]) >= 51 and int(number[0:2]) <= 55):
        print("MASTERCARD")

    elif (len(number) == 13 or len(number) == 16) and int(
        int(number[0:2]) / 10
    ) % 10 == 4:
        print("VISA")

    else:
        print("INVALID")
else:
    print("INVALID")
